### PR TITLE
*: add dev guide

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: dev-guide/build.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Build
+
+## Overview
+
+If you want to dive into TiDB project, change some codes and send us PR, you may follow this guide to build TiDB project.
```
Comment (queenypingcap) [inline dev-guide/build.md]: You can follow this guide to build the TiDB project.

File: dev-guide/build.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+You can see [user build guide](../op-guide/build.md) to check platforms and prerequisites first.
```
Comment (queenypingcap) [inline dev-guide/build.md]: +See [user build guide](../op-guide/build.md) to check the supported platforms and prerequisites first.

File: dev-guide/build.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+If you want to dive into TiDB project, change some codes and send us PR, you may follow this guide to build TiDB project.
+
+You can see [user build guide](../op-guide/build.md) to check platforms and prerequisites first. 
+
+## Install RocksDB
+
+RocksDB 4.9+ is required, you can install RocksDB shared library manually according to [INSTALL.md](https://github.com/facebook/rocksdb/blob/master/INSTALL.md). 
+
+The [build RocksDB script](../scripts/build_rocksdb.sh) can help you build and install RocksDB in system path.
```
Comment (queenypingcap) [inline dev-guide/build.md]: Combine the two paras to one: RocksDB 4.9+ is required. You can install the RocksDB shared library manually according to [INSTALL.md](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) or use the [RocksDB build script](../scripts/build_rocksdb.sh) to build and install RocksDB in the system path.

File: dev-guide/build.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Build TiKV
+
+After you install RocksDB shared library, you can build TiKV directly without `ROCKSDB_SYS_STATIC`.
```
Comment (queenypingcap) [inline dev-guide/build.md]: RocksDB shared library -> the RocksDB shared library

**Comment by siddontang** [comment]
PTAL @queenypingcap @shenli @iamxy @qiuyesuifeng

File: dev-guide/build.md [hunk 5]
```diff
@@ -14,11 +14,59 @@ RocksDB 4.9+ is required, You can install the RocksDB shared library manually ac
...
-```bash
-# cd TiKV source directory
-# build and install the binary in the `bin` directory
-make 
-
-# run test
-make test
-```
\ No newline at end of file
++ Get the TiKV source code
```
Comment (queenypingcap) [inline dev-guide/build.md]: Please add a full stop "." to the end of the sentences.

File: dev-guide/build.md [hunk 6]
```diff
@@ -14,11 +14,59 @@ RocksDB 4.9+ is required, You can install the RocksDB shared library manually ac
...
+    
++ Run test 
+    
+    ```bash
+    make test
+    ```
+
+## Build TiDB
+
++ Make sure set `GOPATH` environment correctly.
```
Comment (queenypingcap) [inline dev-guide/build.md]: ++ Make sure the `GOPATH` environment is set correctly.

File: dev-guide/build.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    ```bash
+    git clone https://github.com/pingcap/tikv.git 
+    ```
++ Enter the source directory to build and install the binary in the `bin` directory.
+
+    ```bash
+    make
+    ```
+    
++ Run test.
```
Comment (shenli) [inline dev-guide/build.md]: Run unit tests?

**Comment by shenli** [comment]
LGTM

**Pull request merged**